go-controller/pkg/util/kube.go

```diff
@@ -16,13 +16,15 @@ import (
 	"k8s.io/client-go/tools/clientcmd"
 	"k8s.io/client-go/util/cert"
 
+	egressipclientset "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/crd/egressip/v1/apis/clientset/versioned"
+
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/cni/types"
 	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
 )
 
 // NewClientset creates a Kubernetes clientset from either a kubeconfig,
 // TLS properties, or an apiserver URL
-func NewClientset(conf *config.KubernetesConfig) (*kubernetes.Clientset, error) {
+func NewClientset(conf *config.KubernetesConfig) (*kubernetes.Clientset, *egressipclientset.Clientset, error) {
```

Comment:
> Looks like your justification for 2 clients is you need to have a client that uses json instead of protobuf

No, we need two clients as to be able to perform object modification on both the v1 objects provided by the kubernetes client set, and the egress IP objects provided by `pkg/crd/egressip/v1/apis/clientset/versioned`. As `EgressIP` is a CRD object, and since those objects are not protobuf serializable (see: https://github.com/kubernetes/kubernetes/issues/55541), we cannot use the same config to initialize that client set as we do for the kubernetes one. But that's not the justification for using two client sets. 

I am not sure I follow on "Just create a generic ovnkube Clientset", you mean having something like:

```
type OVNClientSet struct {
	*kubernetes.Clientset
	eIPClient *egressipclientset.Clientset
}

func NewOVNClientSet(conf *rest.Config) OVNClientSet {
	kClient, _ := kubernetes.NewForConfig(conf)
	eIPClient, _ := egressipclientset.NewForConfig(conf)
	return OVNClientSet{
		kClient,
		eIPClient,
	}
}
```

But that seems more like a cosmetic fix which wouldn't change much more besides the amount of variables this function returns.

Reply:
It sets us up for the future, so that when egress firewall or feature CRD x, y, z, we dont end up with n client sets, or using "egressipclientset" for those other CRDs. We can just have one ovnkube client set which all CRDs can use.

Reply:
Created: https://github.com/ovn-org/ovn-kubernetes/issues/1528 for the next step